compiler/packages/babel-plugin-react-compiler/src/Inference/InferMutationAliasingEffects.ts

```diff
@@ -1340,6 +1350,7 @@ class InferenceState {
             // technically an error, but it's not React specific
             return 'none';
           }
+          case ValueKind.ShallowMutable:
```

Comment:
conditional mutation of a shallow mutable object is also mutation, we'll need to add this to the above case for MutateConditionally/MutateTransitiveConditionally

This function (InferenceState.proto.mutate) is also where we can update the kind if a mutation of a shallow mutable occurs